redash/models/__init__.py

```diff
@@ -204,16 +218,33 @@ def get_by_id(cls, _id):
         return cls.query.filter(cls.id == _id).one()
 
     @classmethod
-    def save_schema(self, schema_info):
+    def save_schema(cls, schema_info):
+        # There was a change in column data.
         if 'columnId' in schema_info:
             ColumnMetadata.query.filter(
-                ColumnMetadata.id == schema_info['columnId'],
                 ColumnMetadata.table_id == schema_info['tableId'],
+                ColumnMetadata.id == schema_info['columnId']
             ).update(schema_info['schema'])
-        else:
-            TableMetadata.query.filter(
+            db.session.commit()
+            return
+
+        if 'sample_queries' in schema_info['schema']:
+            sample_queries = schema_info['schema']['sample_queries']
+            del schema_info['schema']['sample_queries']
```

Comment:
This seems a bit brittle in terms of accessing potentially large in-memory data, could we do this with this instead?

```python
sample_queries = schema_info['schema'].pop('sample_queries', None)
if sample_queries is not None:
    # ...
```

Reply:
Sure we can do that instead, I just want to make sure I understand what your concern is about lots of in-memory data. 

Here's an example of what `schema_info` would look like. It basically just represents the data an admin would input for 1 table:

```
{
    'tableId': 11, 
    'schema': {
      'visible': True,
      'description':  'this is a description',
      'sample_queries': {}
    }
}
```

Do you see this potentially blowing up?

Reply:
Done

Reply:
Wait, I assumed sample_queries could potentially be large blobs of strings, I probably misunderstood the code tbh 😬